staging/src/k8s.io/cloud-provider/app/webhooks.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/server/mux"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/cloud-provider/app/config"
+	"k8s.io/klog/v2"
+)
+
+func init() {
+	_ = corev1.AddToScheme(runtimeScheme)
+	_ = admissionregistrationv1.AddToScheme(runtimeScheme)
+}
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	deserializer  = codecs.UniversalDeserializer()
+)
+
+// WebhooksDisabledByDefault is the webhooks disabled default when starting cloud-controller managers.
+var WebhooksDisabledByDefault = sets.NewString()
+
+type WebhookConfig struct {
+	Path    string
+	Webhook func(*admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error)
+}
+
+type webhookHandler struct {
+	path string
+	http.Handler
+	webhook         func(*admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error)
+	completedConfig *config.CompletedConfig
+	cloud           cloudprovider.Interface
+}
+
+func newWebhookHandlers(webhookConfigs map[string]WebhookConfig, completedConfig *config.CompletedConfig, cloud cloudprovider.Interface) map[string]webhookHandler {
+	webhookHandlers := make(map[string]webhookHandler)
+	for name, config := range webhookConfigs {
+		webhookHandlers[name] = webhookHandler{
```

Comment:
Should we have logic here to "skip" adding any webhooks which have been disabled through the CLI?